src/components/Graph.react.js

```diff
@@ -90,24 +94,63 @@ const RealPlotlyGraph = asyncDecorator(PlotlyGraph, () =>
     LazyLoader.plotly().then(LazyLoader.graph)
 );
 
-class ControlledPlotlyGraph extends PureComponent {
-    render() {
-        return (
-            <Suspense fallback={null}>
-                <RealPlotlyGraph {...this.props} />
-            </Suspense>
-        );
-    }
-}
+const ControlledPlotlyGraph = memo(props => {
+    const {className, id} = props;
+
+    const extendedClassName = className
+        ? 'dash-graph ' + className
+        : 'dash-graph';
+
+    return (
+        <Suspense
+            fallback={
+                <div
+                    id={id}
+                    key={id}
+                    className={`${extendedClassName} dash-graph--pending`}
+                />
+            }
+        >
+            <RealPlotlyGraph
+                {...{
+                    ...props,
+                    className: extendedClassName,
+                }}
```

Comment:
the double `...` looks funny - is this different from
`<RealPlotlyGraph {...props} className={extendedClassName} />`?
Or more explicitly replacing:
`<RealPlotlyGraph {...R.assoc('className', extendedClassName, props)} />`

Reply:
It's the same, will update for cleaner usage